purchases-sample/src/main/java/com/revenuecat/sample/OverviewFragment.kt

```diff
@@ -0,0 +1,91 @@
+        binding = FragmentOverviewBinding.inflate(inflater)
+
+        binding.purchaserInfoCard.setOnClickListener {
+            with(binding.purchaserInfoDetailsContainer) {
+                visibility = if (visibility == View.GONE) View.VISIBLE else View.GONE
+
+                binding.purchaserInfoCardExpandButton
+                    .animate()
+                    .rotationBy(
+                        if (visibility == View.GONE) -ARROW_ANIMATION_ROTATION_DEGREES
+                        else ARROW_ANIMATION_ROTATION_DEGREES
+                    )
+                    .setDuration(resources.getInteger(R.integer.transition_duration).toLong())
+                    .start()
+            }
+        }
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        postponeEnterTransition()
+        view.doOnPreDraw { startPostponedEnterTransition() }
+
+        Purchases.sharedInstance.getPurchaserInfoWith(::showError) { info ->
+            with(binding) {
+                purchaserInfo = info
+
+                purchaserInfoJsonObject.detail = info.jsonObject.toString(JSON_FORMATTER_INDENT_SPACES)
+
+                purchaserInfoActiveEntitlements.detail = formatEntitlements(info.entitlements.active.values)
+                purchaserInfoAllEntitlements.detail = formatEntitlements(info.entitlements.all.values)
+            }
+        }
+
+        Purchases.sharedInstance.getOfferingsWith(::showError, ::populateOfferings)
+    }
+
+    private fun populateOfferings(offerings: Offerings) {
+        binding.overviewOfferingsRecycler.layoutManager = LinearLayoutManager(requireContext())
+        binding.overviewOfferingsRecycler.adapter = OfferingCardAdapter(offerings.all.values.toList(), this)
+    }
+
+    private fun formatEntitlements(entitlementInfos: Collection<EntitlementInfo>): String {
+        return entitlementInfos.joinToString(separator = "\n") { it.toBriefString() }
+    }
+
+    override fun onOfferingClicked(cardView: View, offering: Offering) {
+        exitTransition = MaterialElevationScale(false).apply {
+            duration = requireContext().resources.getInteger(R.integer.transition_duration).toLong()
+        }
+        reenterTransition = MaterialElevationScale(true).apply {
+            duration = requireContext().resources.getInteger(R.integer.transition_duration).toLong()
+        }
+
+        val offeringCardTransitionName = getString(R.string.offering_fragment_transition)
+        val extras = FragmentNavigatorExtras(cardView to offeringCardTransitionName)
+        val directions = OverviewFragmentDirections.actionOverviewFragmentToOfferingFragment(offering)
+        findNavController().navigate(directions, extras)
+    }
+
+    companion object {
+        private const val JSON_FORMATTER_INDENT_SPACES = 4
```

Comment:
Do these need to be inside the companion object? Or can they be outside of the class at the file level? I am not sure actually what's the difference

Reply:
valid point 🤦   i think the only difference is that the companion object associates them with the fragment, which isn't necessarily correct. was just being lazy about creating a constants file, but i think that's the right thing to do.

Reply:
I was actually referring to removing this companion object and doing something like:

```
private const val JSON_FORMATTER_INDENT_SPACES = 4
private const val ARROW_ANIMATION_ROTATION_DEGREES = 180f

class OverviewFragment : Fragment(), OfferingCardAdapter.OfferingCardAdapterListener {
```

Which I think it's equivalent to the constants file. It was more of a question if you knew if there's a difference. 

Now that you say, I think the `JSON_FORMATTER_INDENT_SPACES` should be in another file, since it will probably be used in another fragments right?

Reply:
oh i misread the question, my bad. i think that'd be the same as being in a constants file.

but yeah i think both of these could theoretically be pulled out, if i make the naming of the rotation constant more generic. i'm on it!